src/dtproject/assets/schemas/devcontainers.json

```diff
@@ -1,12 +1,97 @@
-    }
-}
+    },
+    "properties": {
+        "name": {
+            "type": "string",
+            "default": "Duckietown Dev Container",
+            "description": "Name of the Dev Container"
+        },
+        "image": {
+            "type": "string",
+            "default": "${localEnv:DOCKER_REGISTRY:docker.io}/duckietown/${localWorkspaceFolderBasename}:ente-amd64",
+            "description": "Docker image tag"
+        },
+        "workspaceFolder": {
+            "type": "string",
+            "default": "/code/catkin_ws/src/${localWorkspaceFolderBasename}",
+            "description": "Workspace folder path"
+        },
+        "mounts": {
+            "type": "array",
+            "default": [
+                "source=/var/run/avahi-daemon/socket,target=/var/run/avahi-daemon/socket,type=bind",
+                "source=${localWorkspaceFolder},target=/code/src/${localWorkspaceFolderBasename},type=bind",
+                "source=${localWorkspaceFolder}/launchers,target=/launch/${localWorkspaceFolderBasename},type=bind"
+            ],
+            "items": {
+                "type": "string"
+            },
+            "description": "Array of mount configurations"
+        },
+        "runArgs": {
+            "type": "array",
+            "default": [
+                "--net=host",
+                "--rm",
+                "--name",
+                "dts-code-${localWorkspaceFolderBasename}"
+            ],
+            "items": {
+                "type": "string"
+            },
+            "description": "Array of run arguments"
+        },
+        "customizations": {
+            "type": "object",
+            "properties": {
+                "vscode": {
+                    "type": "object",
+                    "properties": {
+                        "extensions": {
+                            "type": "array",
+                            "default": [
+                                "ms-iot.vscode-ros",
+                                "ms-python.python",
+                                "ms-vscode.cpptools"
+                            ],
+                            "items": {
+                                "type": "string"
+                            },
+                            "description": "Array of Visual Studio Code extensions"
+                        }
+                    },
+                    "required": [
+                        "extensions"
+                    ]
+                }
+            },
+            "default": {
+                "vscode": {
+                    "extensions": [
+                        "ms-iot.vscode-ros",
+                        "ms-python.python",
+                        "ms-vscode.cpptools"
+                    ]
+                }
+            },
+            "description": "Customizations for Visual Studio Code"
+        }
+    },    
+    "required": [
+        "name",
+        "image",
+        "workspaceFolder",
+        "mounts",
+        "runArgs"
+    ]
```

Comment:
This block should be moved inside `/patternProperties/.*/`.

Reply:
Also, all the default values should not be at the level of properties inside a devcontainer object but perhaps there should be a default devcontainer object altogether. We need a bit of brainstorming on this.
Perhaps we should extend the devcontainer object specs a bit and include special properties like, `__extends__: List[str]`, `__plain__: bool`, where:

- If you have multiple devcontainer configurations declared, say `A` and `B`, you can do:
```
{
    "A": {
        "image": "duckietown/imageA",
        ...
    }
    "B": {
        "__extends__": ["A"],
        "image": "duckietown/imageB"
    }
}
```
which, in the example above, means that `B` copies `A` but then replaces the image.


- By default, all devcontainer objects are automatically populated with the correct image name, container name, mounts, etc. Internally, this is implemented as having a default devontainer that all devcontainers implicitly extend from. Now, if you define a devcontainer object with the property `__plain__` set to `True`, then this won't happen and you will have to define everything from scratch.

Reply:
Another feature we want is to be able to inject variables into the JSON file. For this, I think we should use `jinja`.